src/libstd/io/stdio.rs

```diff
@@ -872,6 +874,82 @@ impl fmt::Debug for StderrLock<'_> {
     }
 }
 
+/// Equivalent to the [`input`] method except that a prompt
+/// is printed before capturing input. This function returns
+/// either success ([`Ok`]) or failure ([`Err`]) containing
+/// the input. See the [`io::Result`] module documentation
+/// for details.
+///
+/// # Note
+///
+/// If you require more explicit control over capturing
+/// user input, see the [`Stdin::read_line`] method.
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(prompt)]
+/// use std::io::prompt;
+///
+/// fn main() {
+///     let name = prompt("Enter name: ").expect("input failed!");
+///
+///     println!("Your name is {}!", name);
+/// }
+/// ```
+#[unstable(
+    feature = "prompt",
+    reason = "this function may be replaced with a \
+                     more general mechanism",
+    issue = "none"
+)]
+pub fn prompt(prompt: &str) -> Result<String> {
+    stdout().write_all(prompt.as_bytes())?;
+    input()
+}
+
+/// Reads a line from standard input, returning either success
+/// ([`Ok`]) or failure ([`Err`]) containing the input. See the
+/// [`io::Result`] module documentation for details. For
+/// automatic prompt handling, see the [`prompt`] method.
+///
+/// If you need more explicit control over
+/// locking, see the [`Stdin::lock`] and [`Stdout::lock`] methods.
+///
+/// # Note
+///
+/// If you require more explicit control over capturing
+/// user input, see the [`Stdin::read_line`] method.
+///
+/// # Examples
+///
+/// ```no_run
+/// #![feature(input)]
+/// use std::io::input;
+///
+/// fn main() {
+///     print!("Enter name: ");
+///
+///     let name = input().expect("input failed!");
+///
+///     println!("Your name is {}!", name);
+/// }
+/// ```
+#[unstable(
+    feature = "input",
+    reason = "this function may be replaced with a \
+                     more general mechanism",
+    issue = "none"
+)]
+pub fn input() -> Result<String> {
+    stdout().flush()?;
+    let mut input = String::new();
+    match stdin().read_line(&mut input)? {
+        0 => Err(Error::new(ErrorKind::UnexpectedEof, "input reached eof unexpectedly.")),
```

Comment:
```suggestion
        0 => Err(Error::new(ErrorKind::UnexpectedEof, "input reached eof unexpectedly")),
```
Other parts of code suggest does not use `.`